pkg/logcli/client/client.go

```diff
@@ -209,27 +214,39 @@ func (c *DefaultClient) doRequest(path, query string, quiet bool, out interface{
 	}
 
 	var resp *http.Response
-	attempts := c.Retries + 1
+
 	success := false
 
-	for attempts > 0 {
-		attempts--
+	bkcfg := backoff.Config{
+		MinBackoff: time.Duration(c.MinBackoff) * time.Second,
+		MaxBackoff: time.Duration(c.MaxBackoff) * time.Second,
+		MaxRetries: c.MaxRetries,
+	}
+	backoff := backoff.New(context.Background(), bkcfg)
 
+	for {
+		if !backoff.Ongoing() {
+			break
+		}
```

Comment:
I think the backoff check should be at the end like in [promtail](https://github.com/grafana/loki/blob/main/clients/pkg/promtail/client/client.go#L380).

Reply:
No , we have `continue` in this  loop, so if it will be at the end, we have a chance to never reach it.

Reply:
And actually it's a way make line 
https://github.com/grafana/loki/pull/7152/files#diff-4b174cda1ef437ab133e448cff439653c1c84b0244862e942673380f4bdc0e73R227
properly works

we have  1 attempt here to do the requests
when retries are set to 0